### DISCUSSION TITLE
Combine StructuredConfigs and YAML files via composition

### DISCUSSION THREAD

**Discussion by addisonklinke** [mit-ll-responsible-ai/hydra-zen#757] [Q&A] [answered]
On the whole, I strongly agree with Hydra-Zen's philosophy of removing YAML files and building StructuredConfigs directly from code. However, for ML there's a use-case where I _think_ I still want some of the config to live in YAML files. For example, when training a model I'll need an explicit list of feature names (corresponding to columns in the dataset). I don't see a way to do this with StructuredConfigs because
1. There is no reasonable default value since models across different teams use completely different datasets
2. Requiring CLI input each time is infeasible since there can be 100s or 1000s of feature names to type

Here's an example script to demonstrate

```python
# train.py
from dataclasses import dataclass
from typing import List
import hydra
from hydra_zen import builds, ZenStore

immediate_store = ZenStore(deferred_hydra_store=False)


@immediate_store(name="default", group="model")
@dataclass
class Model:
    features: List[str]

    def train(self):
        print(f"Training on features: {self.features}")


@immediate_store(
    name="config",
    hydra_defaults=[
        "_self_",
        {"model": "default"},
    ],
)
def train(model: Model):
    model.train()


@hydra.main(config_path="config", config_name="config", version_base="1.1")
def main(cfg):
    hydra.utils.call(cfg)  # Invokes `_target_` which is `train()`


if __name__ == "__main__":
    main()
```

If `features` was a short list, this would be fine

```
$ python train.py model.features=[f1]
Training on features: ['f1']
```

However when I try to give a longer list with a YAML

```yaml
# config/model.yaml
features:
  - f1
  - f2
  - ...
``` 

`@hydra.main(config_path="config")` isn't picking this up, and so explicit CLI input is still required

```
$ python train.py -h
...
_target_: __main__.train
model:
  features: ???  # <- Why can't I get this to populate?
  _target_: __main__.Model
```

I also tried renaming `config/model.yaml -> config/config.yaml` to align directly with the `config_path` param of `@hydra.main()`, but that runs into the following 

```
hydra.errors.ConfigCompositionException: 'config' is validated against ConfigStore schema with the same name.
This behavior is deprecated in Hydra 1.1 and will be removed in Hydra 1.2.
In addition, the automatically matched schema contains a defaults list.
This combination is no longer supported.
See https://hydra.cc/docs/1.2/upgrades/1.0_to_1.1/automatic_schema_matching for migration instructions.
```

which makes it appear Hydra assumes I want the StructuredConfig to validate the schema of my YAML, when in fact I want to compose the StructuredConfig and YAML together. Surely there's got to be a way to do this?

**Comment by addisonklinke** [comment, marked as answer]
Found the solution for my initial problem: refactor `config/model.yaml` to `config/model/default.yaml`. Then the `{"model": "default"}` entry of Hydra's defaults list will pick this up. The previous structure would correspond to a top-level `@hydra.main(config_name="model", ...)` which was not my intent

**Reply by jgbos** [reply]
that's the only solution I know of too off the top of my head.  The standard structure of config files is that directories are groups.  The other solution is to do `--config-name model`
